### PR TITLE
Add schema on structure and values

### PR DESCRIPTION
**Pull request by tobi**
using cue to create a deep schema for the .json files

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,6 @@
-/shopify
+.byebug_history
+schema/attribute_data.cue
+
 /_site
 /docs/_data
-.byebug_history
+/shopify
```
Comment (Chris911) [inline .gitignore:6]: I don't see this folder in `main` or the `gh-pages` branch. Still needed?

File: dev/bin/vet_schema [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
```
Comment (Chris911) [inline dev/bin/vet_schema:1]: Nit: should the script be named `vet_dist_files` instead of `vet_schema`?

File: schema/categories_schema.cue [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package shopify
+
+#categoryId: =~"^gid://shopify/Taxonomy/Category/\\w+(-\\d+)*$"
```
Comment (Chris911) [inline schema/categories_schema.cue:3]: Nit: `\w+` will include underscore and digits. If we want to be stricter we can go with: `[a-z]{2}`. Also use for `prefix` in `verticals`.

File: schema/categories_schema.cue [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+// Structure for categories, including recursive definitions for children, attributes, and ancestors
+#category: {
+	id:        string & #categoryId
+	name:      string
+	level:     int & >=0
+	full_name: string
+	parent_id: *null | string & #categoryId
+	attributes: [...#attribute_ref]
+	children: [...#category_ref] | *[]
```
Comment (Chris911) [inline schema/categories_schema.cue:32]: Not sure the `| *[]` is necessary as we always have a `children`/`ancestors` key. Validation passes locally without it.

File: schema/attributes_schema.cue [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package shopify
+
+// Structure for attributes, including recursive definitions for values
+#attribute: {
+	id:   =~"^gid://shopify/Taxonomy/Attribute/\\d+$"
+	name: string
+	values: [...{
+		// id:   "\(id)/\\d+" // CUE can't handle nested checks with regexes yet
```
Comment (Chris911) [inline schema/attributes_schema.cue:8]: This works locally with `#AttributeGid` defined as a constant. `\()` is used for [interpolation](https://cuelang.org/docs/tutorials/tour/expressions/interpolation/).

```
package shopify

#AttributeGid: "gid://shopify/Taxonomy/Attribute/\\d+"

// Structure for attributes, including recursive definitions for values
#attribute: {
	id:   =~"^\(#AttributeGid)$"
	name: string
	values: [...{
		id:   =~"^\(#AttributeGid)/\\d+$"
		name: string
	}]
}

version: =~"^\\d+.\\d+.\\d+$"
attributes: [...#attribute]
```

File: .gitignore (hunk 1 above)
Comment (elsom25) [reply .gitignore:6]: Ah this was here before, but looks to be from the early days. Will 🔥

**Pull request merged**